tracer/src/Datadog.Tracer.Native/iast/dataflow.cpp

```diff
@@ -649,6 +649,7 @@ HRESULT Dataflow::RewriteMethod(MethodInfo* method, ICorProfilerFunctionControl*
                         auto result = InstrumentInstruction(rewriter, pInstr, moduleAspectRefs);
                         pInstr = result.instruction;
                         written |= result.written;
+                        ProcessIastMetric(result);
```

Comment:
This should go inside `InstrumentInstruction`, where you have access to the aspect being applied and thus, don't need to change the `InstrumentResult `struct.